### PR TITLE
Rad-2066 Spring Security fix

### PR DESCRIPTION
**Pull request by MertCingoz**

### CODE REVIEW

File: Core/src/com/serotonin/m2m2/Common.java [hunk 1]
```diff
@@ -453,6 +453,14 @@ public static PermissionHolder getUser() throws PermissionException {
...
 
+    public static PermissionHolder getPermissionHolder() {
+        PermissionHolder permissionHolder = null;
+        try {
+            permissionHolder = getUser();
+        } catch (PermissionException ignored) {}
+        return permissionHolder;
+    }
+
     public static Path getTempPath() {
```
Comment (jazdw) [inline Core/src/com/serotonin/m2m2/Common.java:464]: I would avoid adding this method, we should always have a PermissionHolder available. If we are trying to find out the current user and there isn't one, its an error and we want to know about it.

**Review by jazdw — CHANGES_REQUESTED** [review]

File: Core/src/com/serotonin/m2m2/Common.java (hunk 1 above)
Comment (MertCingoz) [reply Core/src/com/serotonin/m2m2/Common.java]: @jazdw can you check my last 2 commit,

Looks like using strategy `MODE_THREADLOCAL` is removing authentication for the child threads.
If task is created from a child thread, it doesn't have the user available (Authentication is null).
I have fixed the tests by adding admin user but also had to change strategy to `MODE_INHERITABLETHREADLOCAL`

So tests that are using to create tasks from child thread or even JUnit timeout tests have no user available.

Tests are fixed but do we need change default strategy to `MODE_INHERITABLETHREADLOCAL`?
If we have some tasks in real application which created in child thread(s) will throw error as well.

Or should we ignore the user and use null Authentication for this child threads as before ?

Comment (jazdw) [reply Core/src/com/serotonin/m2m2/Common.java]: > Tests are fixed but do we need change default strategy to MODE_INHERITABLETHREADLOCAL?
I don't think we should be setting the mode to inheritable for the tests or for the real application.

The idea is that `MODE_THREADLOCAL` will not implicitly propagate the security context to new threads. You must explicitly set the security context in any new threads which are spawned. We need to figure out where these new threads are coming from and do something like `DelegatingSecurityContextExecutorService` or `Task` which explicitly set the security context in the new thread. Lets get on a call today and discuss.

**Comment by terrypacker** [comment]
@MertCingoz I'm wondering if you noticed that the security context is not set in the MockMangoLifecycle.  In `com.serotonin.m2m2.Lifecycle` (ma-core-prive) 

we have this code:
```
                // set the authentication for the main thread's security context, enables the use of services requiring a user in the
                // module lifecycle hooks
                SecurityContext securityContext = SecurityContextHolder.getContext();
                securityContext.setAuthentication(new     PreAuthenticatedAuthenticationToken(PermissionHolder.SYSTEM_SUPERADMIN, null));
```

I think this should be in the initialize method of MockMangoLifecycle...  I tested this and it doesn't fix the persistent test failures but might have an impact on the other tests.

**Comment by MertCingoz** [comment]
> I think this should be in the initialize method of MockMangoLifecycle... I think you should review this with the changes you made in the persistent TCP tests too.

@terrypacker `MangoTestBase.staticSetup` should be enough to set admin user for the tests. It is already calling `setSuperadminAuthentication` method.

But JUnit timeout tests running on seperate thread which means `@BeforeClass` or `@Before` decorators are not sufficent to set security context when we use `MODE_THREADLOCAL` strategy

So `lifecycle.initialize` is called on `@Before` only for first test which doesn't change anything for timeout tests at all

Can you check new test cases
- TimerTaskPermissionHolderTest.timeoutTestNoContext
- TimerTaskPermissionHolderTest.timedTestHasContext

**Review by jazdw — APPROVED** [review]

**Pull request merged**